sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java

```diff
@@ -330,21 +301,41 @@ public void onFailure(Throwable e) {
         return indexNames;
     }
 
-    private static class SettableFutureToNullActionListener<T> implements ActionListener<T> {
-        private final SettableFuture<?> future;
+    private static boolean areAllMappingsEqual(MetaData metaData, String... indices) {
+        Map<String, Object> lastMapping = null;
+        for (String index : indices) {
+            try {
+                Map<String, Object> mapping = metaData.index(index).mapping(Constants.DEFAULT_MAPPING_TYPE).getSourceAsMap();
+                if (lastMapping != null && !lastMapping.equals(mapping)) {
+                    return false;
+                }
+            } catch (Throwable t) {
+                Throwables.propagate(t);
+            }
+        }
+        return true;
+    }
+
+    private static class OperationsAwareActionListener<T> implements ActionListener<T> {
+
+        private final SettableFuture<Long> result;
+        private final AtomicInteger operations;
 
-        public SettableFutureToNullActionListener(SettableFuture<?> future) {
-            this.future = future;
+        public OperationsAwareActionListener(SettableFuture<Long> result, AtomicInteger operations) {
+            this.result = result;
+            this.operations = operations;
         }
 
         @Override
-        public void onResponse(T response) {
-            future.set(null);
+        public void onResponse(T putMappingResponse) {
+            if (operations.decrementAndGet() == 0) {
+                result.set(1L);
+            }
         }
 
         @Override
         public void onFailure(Throwable e) {
-            future.setException(e);
+            result.setException(e);
```

Comment:
same like before, they can't be stopped so they will keep running but I don't see a need to wait for them